homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -408,3 +431,364 @@ def media_position_updated_at(self):
+        """
+        # If we should join the group, which is served by the main zone, we can simply select main_sync as input.
+        _LOGGER.debug("%s called service client join", self.entity_id)
+        if self.state == STATE_OFF:
+            await self.async_turn_on()
+        if self.ip_address == server.ip_address:
+            if server.zone == DEFAULT_ZONE:
+                await self.async_select_source(ATTR_MAIN_SYNC)
+                server.async_write_ha_state()
+                return
+
+            # It is not possible to join a group hosted by zone2 from main zone.
+            raise Exception("Can not join a zone other than main of the same device.")
+
+        if self.musiccast_zone_entity.is_server:
+            # If one of the zones of the device is a server, we need to unjoin first.
+            _LOGGER.info(
+                "%s is a server of a group and has to stop distribution "
+                "to use MusicCast for %s",
+                self.musiccast_zone_entity.entity_id,
+                self.entity_id,
+            )
+            await self.musiccast_zone_entity.async_server_close_group()
+
+        elif self.is_client:
+            if self.coordinator.data.group_id == server.coordinator.data.group_id:
+                _LOGGER.warning("%s is already part of the group", self.entity_id)
+                return
+
+            _LOGGER.info(
+                "%s is client in a different group, will unjoin first",
+                self.entity_id,
+            )
+            await self.async_client_leave_group()
+
+        elif (
+            self.ip_address in server.coordinator.data.group_client_list
+            and self.coordinator.data.group_id == server.coordinator.data.group_id
+            and self.coordinator.data.group_role == "client"
+        ):
+            # The device is already part of this group (e.g. main zone is also a client of this group).
+            # Just select mc_link as source
+            await self.async_select_source(ATTR_MC_LINK)
+            # As the musiccast group has changed, we need to trigger the servers ha state.
+            # In other cases this happens due to the callback after the dist updated message.
+            server.async_write_ha_state()
+            return
+
+        _LOGGER.debug("%s will now join as a client", self.entity_id)
+        await self.coordinator.musiccast.mc_client_join(
+            server.ip_address, group_id, self._zone_id
+        )
+
+        # Ensure that mc link is selected. If main sync was selected previously, it's possible that this does not
+        # happen automatically
+        await self.async_select_source(ATTR_MC_LINK)
+
+    async def async_client_leave_group(self, force=False):
+        """Make self leave the group.
+
+        Should only be called for clients.
+        """
+        _LOGGER.debug("%s client leave called", self.entity_id)
+        if not force and (
+            self.source == ATTR_MAIN_SYNC
+            or len(
```

Comment:
In Python we don't need to check the length of containers to check if they are not empty. It's enough to just do:
```py
if container:
```

Please clean this up where we use `len`.